plugins/scaffolder/src/components/TemplateEditorPage/TemplateEditorForm.tsx

```diff
@@ -123,7 +123,9 @@ export function TemplateEditorForm(props: TemplateEditorFormProps) {
           setSteps(undefined);
           return;
         }
-        const parsed: JsonValue = yaml.parse(content);
+        const parsed: JsonValue = yaml
+          .parseAllDocuments(content)
```

Comment:
you also should add a `.filter(c => c)` to get rid of spurious null documents. Like we do for example [here](https://github.com/backstage/backstage/blob/abdece5ae0ed71516a905caff935153d180e6acb/plugins/catalog-backend/src/modules/core/PlaceholderProcessor.ts#L177).

Reply:
okay but same is required in src/next/ ?

Reply:
Anything that calls `parseAllDocuments` should have that check.

You also moved the `.filter` AFTER `.toJSON`; at that point it's already too late; you want to filter directly after `parseAllDocuments` because that's the one that may have nulls in its response array entries

Reply:
brilliant